frontend/src/api/RestaurantApi.tsx

```diff
@@ -16,6 +16,7 @@ export const useSearchRestaurants = (
     const params = new URLSearchParams();
     params.set('searchQuery', searchState.searchQuery);
     params.set('page', searchState.page.toString());
+    params.set('selectedCuisines', searchState.selectedCuisines.join(','));
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/41#issuecomment-2407615424>#dd91ac</a><span> - 10/11/2024, 03:13 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The current implementation doesn't encode the selectedCuisines values, which could lead to issues with special characters in cuisine names.
</div>
<div id="fix">
<b>Fix:</b> Use encodeURIComponent() to properly encode each cuisine name before joining them.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -19 +19 @@
 -    params.set('selectedCuisines', searchState.selectedCuisines.join(','));
 +    params.set('selectedCuisines', searchState.selectedCuisines.map(cuisine => encodeURIComponent(cuisine)).join(','));
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged